game/ui/gamemenu.cpp

```diff
@@ -670,36 +676,52 @@ void GameMenu::getText(const Item& it, std::vector<char> &out) {
     out.resize(1);
   out[0]='\0';
 
-  const auto& src = it.handle.text[0];
-  if(it.handle.type==Daedalus::GEngineClasses::C_Menu_Item::MENU_ITEM_TEXT) {
+  const auto& src = it.handle->text[0];
+  if(it.handle->type==phoenix::c_menu_item_type::text) {
     size_t size = std::strlen(src.c_str());
     out.resize(size+1);
     std::memcpy(out.data(),src.c_str(),size+1);
+
+    if (it.handle->flags & phoenix::c_menu_item_flags::multiline) {
```

Comment:
That is, because ZenLib pre-processes the strings coming from within scripts while phoenix doesn't. ZenLib does effectively the same thing in the `ZString` constructor here:

https://github.com/Try/ZenLib/blob/65b2c1f34e3d3dcd4dd6e6e1c9526996f76e0fe3/daedalus/ZString.cpp#L46-L87

Which is called from within `DatFile::readSymTable` here:

https://github.com/Try/ZenLib/blob/65b2c1f34e3d3dcd4dd6e6e1c9526996f76e0fe3/daedalus/DATFile.cpp#L58-L62

Reply:
Preprocessing look like more correct approach, isn't it?
Also handling `\n`, at draw-time means that concatenated strings, like "str1" + '\\" + "n str2" wont produce same output on the screen. Basically my question is: are you sure that this is correct behavior, from engine side?

Reply:
I know that un-escaping strings is not generally correct for all files. Animations, for example, contain unescaped `\` characters in Windows file paths. I am, however, changing it in `phoenix::script` since there were apparently no issues with `ZenLib` doing it. If we find a place where un-escaping causes issues, we'll have to re-evaluate though.